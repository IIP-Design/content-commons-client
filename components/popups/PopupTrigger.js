import React, { Component } from 'react';
import { Popup, Button } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import '../../styles/tooltip.scss';

class PopupTrigger extends Component {
  constructor( props ) {
    super( props );

    this.state = {
      isMobile: false,
    };
  }

  componentDidMount() {
    this.isMobile();
  }

  isMobile = () => {
    if ( window.outerWidth < 600 ) {
      this.setState( { isMobile: true } );
    }
  };


  render() {
    // const trigger = withTooltip( <Button>my button </Button> );
    return (
      <span style={ { display: this.props.show ? 'inline-block' : 'none' } }>
        <Popup
          trigger={ (
            <Button className="trigger" tooltip={ this.props.toolTip }>
              <img
                src={ this.props.icon.img }
                width={ this.props.icon.dim }
                height={ this.props.icon.dim }
                alt={ this.props.toolTip }
              />
              { ' ' }
            </Button>
          ) }
          on="click"
          onOpen={ this.handleOnOpen }
          onClose={ this.handleOnClose }
          className={ !this.state.isMobile ? 'popupElem_wrapper' : 'popupElem_wrapper popupElem_wrapper--mobile' }
          content={ this.props.content }
          offset={ 4 }
          position={ this.state.isMobile ? 'bottom center' : 'bottom right' }
        />
      </span>
    );
  }
}

PopupTrigger.propTypes = {
  toolTip: PropTypes.string,
  icon: PropTypes.object,
  content: PropTypes.node,
  show: PropTypes.bool,
};

export default PopupTrigger;
