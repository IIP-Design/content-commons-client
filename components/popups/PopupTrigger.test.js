import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import PopupTrigger from './PopupTrigger';

const props = {
  toolTip: 'Share Project',
  icon: { img: 'shareIcon.svg', dim: 18 },
  content: <div>content</div>,
  show: true,
};

const Component = <PopupTrigger { ...props } />;

describe( '<PopupTrigger />', () => {
  it( 'renders without crashing', () => {
    const wrapper = mount( Component );

    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'sets initial state', () => {
    const wrapper = mount( Component );
    const inst = wrapper.instance();

    expect( inst.state ).toEqual( { isMobile: false } );
  } );

  it( 'componentDidMount calls isMobile', () => {
    const wrapper = mount( Component );
    const inst = wrapper.instance();
    const spy = jest.spyOn( inst, 'isMobile' );

    inst.componentDidMount();
    expect( spy ).toHaveBeenCalled();
  } );

  it( 'isMobile sets isMobile to true in state if window.outerWidth is < 600', () => {
    const wrapper = mount( Component );
    const inst = wrapper.instance();

    // initial state
    expect( inst.state.isMobile ).toEqual( false );

    const windowOuterWidth = window.outerWidth;

    window.outerWidth = 500;
    inst.isMobile();
    expect( window.outerWidth ).toEqual( 500 );
    expect( inst.state.isMobile ).toEqual( true );

    // restore default 1024 to avoid affecting next test
    window.outerWidth = windowOuterWidth;
    expect( window.outerWidth ).toEqual( 1024 );
  } );

  it( 'isMobile does not set isMobile to true in state if window.outerWidth is > 600', () => {
    const wrapper = mount( Component );
    const inst = wrapper.instance();

    // initial state
    expect( inst.state.isMobile ).toEqual( false );

    inst.isMobile();
    expect( window.outerWidth ).toEqual( 1024 );
    expect( inst.state.isMobile ).toEqual( false );
  } );

  it( 'sets inline display value to none if !props.show', () => {
    const wrapper = mount( Component );

    wrapper.setProps( { show: false } );
    const span = wrapper.find( 'span' );

    expect( span.prop( 'style' ) ).toEqual( { display: 'none' } );
  } );

  it( 'sets correct Popup position and className props', () => {
    const wrapper = mount( Component );
    const popup = () => wrapper.find( 'Popup' );

    expect( wrapper.state( 'isMobile' ) ).toEqual( false );
    expect( popup().prop( 'className' ) ).toEqual( 'popupElem_wrapper' );
    expect( popup().prop( 'position' ) ).toEqual( 'bottom right' );

    wrapper.setState( { isMobile: true } );
    expect( wrapper.state( 'isMobile' ) ).toEqual( true );
    expect( popup().prop( 'className' ) )
      .toEqual( 'popupElem_wrapper popupElem_wrapper--mobile' );
    expect( popup().prop( 'position' ) ).toEqual( 'bottom center' );
  } );
} );
