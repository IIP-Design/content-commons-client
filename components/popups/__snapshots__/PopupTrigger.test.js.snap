// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopupTrigger /> renders without crashing 1`] = `
<PopupTrigger
  content={
    <div>
      content
    </div>
  }
  icon={
    Object {
      "dim": 18,
      "img": "shareIcon.svg",
    }
  }
  show={true}
  toolTip="Share Project"
>
  <span
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <Popup
      className="popupElem_wrapper"
      content={
        <div>
          content
        </div>
      }
      disabled={false}
      eventsEnabled={true}
      offset={4}
      on="click"
      pinned={false}
      popperModifiers={Array []}
      position="bottom right"
      trigger={
        <Button
          as="button"
          className="trigger"
          tooltip="Share Project"
        >
          <img
            alt="Share Project"
            height={18}
            src="shareIcon.svg"
            width={18}
          />
           
        </Button>
      }
    >
      <Portal
        closeOnDocumentClick={true}
        closeOnEscape={true}
        closeOnTriggerClick={true}
        eventPool="default"
        onClose={[Function]}
        onMount={[Function]}
        onOpen={[Function]}
        onUnmount={[Function]}
        openOnTriggerClick={true}
        trigger={
          <Button
            as="button"
            className="trigger"
            tooltip="Share Project"
          >
            <img
              alt="Share Project"
              height={18}
              src="shareIcon.svg"
              width={18}
            />
             
          </Button>
        }
        triggerRef={
          Object {
            "current": <button
              class="ui button trigger"
              tooltip="Share Project"
            >
              <img
                alt="Share Project"
                height="18"
                src="shareIcon.svg"
                width="18"
              />
               
            </button>,
          }
        }
      >
        <Ref
          innerRef={[Function]}
        >
          <RefFindNode
            innerRef={[Function]}
          >
            <Button
              as="button"
              className="trigger"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              tooltip="Share Project"
            >
              <Ref
                innerRef={
                  Object {
                    "current": <button
                      class="ui button trigger"
                      tooltip="Share Project"
                    >
                      <img
                        alt="Share Project"
                        height="18"
                        src="shareIcon.svg"
                        width="18"
                      />
                       
                    </button>,
                  }
                }
              >
                <RefFindNode
                  innerRef={
                    Object {
                      "current": <button
                        class="ui button trigger"
                        tooltip="Share Project"
                      >
                        <img
                          alt="Share Project"
                          height="18"
                          src="shareIcon.svg"
                          width="18"
                        />
                         
                      </button>,
                    }
                  }
                >
                  <button
                    className="ui button trigger"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    tooltip="Share Project"
                  >
                    <img
                      alt="Share Project"
                      height={18}
                      src="shareIcon.svg"
                      width={18}
                    />
                     
                  </button>
                </RefFindNode>
              </Ref>
            </Button>
          </RefFindNode>
        </Ref>
      </Portal>
    </Popup>
  </span>
</PopupTrigger>
`;
